src/compile.js

```diff
@@ -511,6 +511,37 @@ Compiler.prototype.cyield = function(e) {
     return "$gen.gi$sentvalue"; // will either be none if none sent, or the value from gen.send(value)
 };
 
+Compiler.prototype.cyieldfrom = function (e) {
```

Comment:
Overall: This function is huge and complex and has no comments! Something explaining what we're doing here and how would have helped (as it was I got @s-cork on a call to walk me through it, and I am now convinced, but not everyone can do that ;) ).

Something like (to my understanding):
```javascript
// First, get the iterator.
// [afterIter block starts]
// Check if we've been sent a value, because if so we need to send it on into the iterator by calling its `send()` method.
// If that `send()` method raises `StopIteration`, the sent value needs to be redirected to us (as it hasn't been used yet).
// As a performance optimisation, if we're sending to a generator, we can pass it into `tp$iternext()` and it will do this all for us (see generator.js).
// Either way, if the iterator is done (undefined) and we still have an unused sent value, it will be in `[iterable].$value`, so we grab it from there and move on from the `yield from` ("afterBlock")
// Otherwise, we return the next value from the iterator, and set the generator to do this all again to get the next element.
```

Reply:
Added comments and some formatting to make the code more readable